spark3/src/main/java/org/apache/iceberg/spark/FileRewriteCoordinator.java

```diff
@@ -71,6 +72,8 @@ public void commitRewrite(Table table, Set<String> fileSetIDs) {
     table.newRewrite()
         .rewriteFiles(rewrittenDataFiles, newDataFiles)
         .commit();
+
+    fileSetIDs.stream().map(id -> toID(table, id)).forEach(resultMap::remove);
```

Comment:
The `commitRewrite` methods are only called from tests. Since this has changed quite a bit, have we looked to see what is still needed?

Reply:
As noted above I think we'll do a separate PR to clean up this file post merge.